libs/coin-framework/src/account/accountId.ts

```diff
@@ -11,6 +11,43 @@ function ensureNoColon(value: string, ctx: string): string {
   return value;
 }
 
+export function safeEncodeTokenId(tokenId: string): string {
+  if (!tokenId) return "";
+
+  const URIEncoded = encodeURIComponent(tokenId);
+  const dashUnderscoreSafe = URIEncoded.replace(/-/g, "~!dash!~")
+    .replace(/_/g, "~!underscore!~")
+    .replace(/\+/g, "~!plus!~");
+
+  invariant(
+    !dashUnderscoreSafe.includes("-") &&
+      !dashUnderscoreSafe.includes("_") &&
+      !dashUnderscoreSafe.includes("+"),
+    `TokenId ${dashUnderscoreSafe} shouldn't contain - or _`,
```

Comment:
you can remove this invariant. the code before it makes this case impossible. that way we can optim a bit the perf (includes can be costly as i assume we call this function a lot)

Reply:
Alright, done 👌